src/RepoDriver.sol

```diff
@@ -0,0 +1,530 @@
+    /// See `requestUpdateOwner` for more details.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @param fee The fee in Link to pay for the request.
+    /// @return userId The ID of the user.
+    function _requestUpdateOwner(Forge forge, bytes memory name, uint256 fee)
+        internal
+        returns (uint256 userId)
+    {
+        RepoDriverAnyApiStorage storage storageRef = _repoDriverAnyApiStorage();
+        address operator = address(storageRef.operator);
+        require(operator != address(0), "Operator address not set");
+        uint256 nonce = storageRef.nonce++;
+        bytes32 requestId = keccak256(abi.encodePacked(this, nonce));
+        userId = calcUserId(forge, name);
+        storageRef.requestedUpdates[requestId] = userId;
+        bytes memory payload = _requestPayload(forge, name);
+        bytes memory callData = abi.encodeCall(
+            OperatorInterface.operatorRequest,
+            (
+                address(0), // ignored, will be replaced in the operator with this contract's address
+                0, // ignored, will be replaced in the operator with the fee
+                storageRef.jobId,
+                this.updateOwnerByAnyApi.selector,
+                nonce,
+                2, // data version
+                payload
+            )
+        );
+        require(linkToken.transferAndCall(operator, fee, callData), "Transfer and call failed");
+        // slither-disable-next-line reentrancy-events
+        emit OwnerUpdateRequested(userId, forge, name);
+    }
+
+    /// @notice Builds the AnyApi generic `bytes` fetching request payload.
+    /// It instructs the operator to fetch the current owner of the user.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return payload The AnyApi request payload.
+    function _requestPayload(Forge forge, bytes memory name)
+        internal
+        view
+        returns (bytes memory payload)
+    {
+        // slither-disable-next-line uninitialized-local
+        BufferChainlink.buffer memory buffer;
+        buffer = BufferChainlink.init(buffer, 256);
+        buffer.encodeString("get");
+        buffer.encodeString(_requestUrl(forge, name));
+        buffer.encodeString("path");
+        buffer.encodeString(_requestPath());
+        return buffer.buf;
+    }
+
+    /// @notice Builds the URL for fetch the `FUNDING.json` file for the given repository.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return url The built URL.
+    function _requestUrl(Forge forge, bytes memory name)
+        internal
+        pure
+        returns (string memory url)
+    {
+        if (forge == Forge.GitHub) {
+            return string.concat(
+                "https://raw.githubusercontent.com/", string(name), "/HEAD/FUNDING.json"
```

Comment:
Other forges can't be added without an update because it involves not only a new URL but also a new user ID calculation logic. We'd need to implement something like a diamond or a registry of contracts per forge ID to allow adding new forges without touching the core.

No URL updates probably will be necessary in years, and we're already planning switching to Chainlink functions. All URLs and paths will be outsourced to JS code anyway, the admins will probably be able to switch it to a different one when there's a need for a URL change. I'd keep it simple for now.